### DISCUSSION TITLE
v2.0.6 Patch Release

### DISCUSSION THREAD

**Discussion by Wenn0101** [sparkfun/Arduino_Apollo3#368] [General]
**Release Candidate is up for v2.0.6:**
Patch release attempting to address a lot of issues that have been annoying users.

**Changelog:**
-Re-add MicroMod Support to v2 core (https://github.com/sparkfun/Arduino_Apollo3/issues/343)
-Fix permissions issue on macOS and Linux (https://github.com/sparkfun/Arduino_Apollo3/issues/330)
-Fix svl uploader for bigSur (https://github.com/sparkfun/Arduino_Apollo3/issues/299)
-Fix AnalogRead reporting wrong pin (https://github.com/sparkfun/Arduino_Apollo3/issues/303) 
-Diversify BLE MAC addresses by default (prevent duplicate MAC addresses) (https://github.com/sparkfun/Arduino_Apollo3/issues/290)
-Fix non-compiling examples for I2C and SPI (https://github.com/sparkfun/Arduino_Apollo3/issues/350)
-Fix Arduino API issues (for example including IPAddress.h should now work) (no active issue)

**Changes since discussion creation:**
(none)

**How to test:**
- navigate to your directory with the latest version
- rename the directory from 2.0.5 to .bckup2.0.5 (or simply delete it and download it later if wanted)
- clone this repo
`git clone https://github.com/sparkfun/Arduino_Apollo3.git`
- checkout release-candidate branch
`cd Arduino_Apollo3`
`git checkout release-candidate`
- update all submodules
`git submodule update --init --recursive`
- rename directory from Arduino_Apollo3 to 2.0.5
- restart Arduino and enjoy.

**Comment by adamgarbo** [comment]
Thanks, @Wenn0101!

Looking forward to many of these fixes. I'll kick the tires and let you know if I find any gremlins. When is 2.0.6 set to release?

Cheers,
Adam

**Reply by adamgarbo** [reply]
From initial testing, it appears that the macOS Big Sur and permission issues are now resolved. I was able to successfully upload to the MicroMod Artemis Processor using the updated Core. 

I did identify that there is a bug that is preventing SdFat library examples from compiling, but this also applies to v2.0.5, and the issue appears to be related to Mbed's version of SPI. I created a new GitHub issue about this.

Also a quick note that before executing `git checkout release-candidate`, there should be a `cd Arduino_Apollo3` step. Otherwise, users will run into the error:

```
(base) adam@Adams-MacBook-Air apollo3 % git checkout release-candidate
fatal: not a git repository (or any of the parent directories): .git
```

**Reply by Wenn0101** [reply]
Thanks for the bug report, I will look into it.

You are correct there should be a step to cd into Arduino_Apollo3. Updating the instructions now.

**Comment by paulvha** [comment]
some feedback on BLE. Good to see this works with different BDaddr now. I have tested on Edge and ATP board.
The permission on bootloaders worked good without a need to changed

With respect to the **remark on line 199**
```
// Worth noting, these can also be set in your mbed_app.json file, with the following
//
//      "target.macros_add":["AM_CUSTOM_BDADDR=1", "AM_CUSTOM_BDADDR_TEMPLT1=0xff00aa"]
```
Please be aware that as you build a library from Mbed-OS, mbed_app.json is NOT included in the build process. (build.py only in build_api.py)

**Remark on  line 208**
`//     mbed compile -t GCC_ARM -m SFE_ARTEMIS_THING_PLUS -D AM_CUSTOM_BDADDR -D ARDUINO`

Shouldn't that include the --library  and --source options :
`// mbed compile --library --source=mbed-os  -m SFE_ARTEMIS_ATP -t GCC_ARM  -D AM_CUSTOM_BDADDR -D ARDUINO`

regards,
Paul

**Reply by Wenn0101** [reply]
I will add another comment here that includes instructions for building the library. Good point.

**Comment by marsfan** [comment]
What does the _Legacy Arduino API Support_ change mean?

**Reply by Wenn0101** [reply]
This is a good question.
A submodule that we include is the [ArduinoCore-API](https://github.com/arduino/ArduinoCore-API). In this submodule is a deprecated folder. We added this deprecated folder the the list of include paths.

There are a bunch of examples that have things like `#include IPAddress.h`. From what I can tell reading the arduino documentation, you are no longer supposed to include these because they all come with Arduino.h.

But in order to support examples that do it this older way, we updated the build process to include this "deprecated" folder. These examples should now compile and link correctly as a result.

**Reply by marsfan** [reply]
Ah. So its really just for people who have been working with Arduino for a really long time and are used to old conventions. Cool.
